pkg/libvmi/vm.go

```diff
@@ -76,8 +76,27 @@ func WithDataVolumeTemplate(datavolume *v1beta1.DataVolume) VMOption {
 	}
 }
 
+func resourcesRemovedFromVMI(vmiSpec *v1.VirtualMachineInstanceSpec) {
+	vmiSpec.Domain.CPU = nil
+	vmiSpec.Domain.Memory = nil
+	vmiSpec.Domain.Resources = v1.ResourceRequirements{}
+}
+
+func preferencesRemovedFromVMI(vmiSpec *v1.VirtualMachineInstanceSpec) {
+	vmiSpec.TerminationGracePeriodSeconds = nil
+	vmiSpec.Domain.Features = &v1.Features{}
+	vmiSpec.Domain.Machine = &v1.Machine{}
```

Comment:
does it make a difference if they are an empty struct instead of a nil?

Reply:
Ah nice catch, these are optional and should be `nil`, I'll fix now.

FWIW this is commit is part of https://github.com/kubevirt/kubevirt/pull/11609 if you could review that PR once I've fixed this?